src/sections/create-dataset/HostCollectionForm/HostCollectionForm.tsx

```diff
@@ -0,0 +1,45 @@
+import { Button, Col, Row } from '@iqss/dataverse-design-system'
+import { Form } from '@iqss/dataverse-design-system'
+import { useTranslation } from 'react-i18next'
+import { useNotImplementedModal } from '../../not-implemented/NotImplementedModalContext'
+import { NotImplementedModal } from '../../not-implemented/NotImplementedModal'
+
+interface HostCollectionFormProps {
+  collectionId: string
+}
+
+export function HostCollectionForm({ collectionId }: HostCollectionFormProps) {
+  const { t } = useTranslation('createDataset')
+  const { hideModal, isModalOpen, showModal } = useNotImplementedModal()
+  const onClick = () => {
+    showModal()
+  }
+  return (
+    <>
+      <NotImplementedModal show={isModalOpen} handleClose={hideModal} />
```

Comment:
I think `<NotImplementModal />` should be in` CreateDataset.tsx`, since it is listening for context state changes it does not need to be in the component that is opening it.
```
  return (
    <>
      <article>
        <header>
          <h1>{t('pageTitle')}</h1>
        </header>
        <SeparationLine />
        <HostCollectionForm collectionId={collectionId} />

        {isLoadingMetadataBlocksConfiguration ? (
          <DatasetFormSkeleton />
        ) : (
          <DatasetForm
            repository={repository}
            collectionId={collectionId}
            metadataBlocks={metadataBlocks}
            formDefaultValues={formDefaultValues}
            errorLoadingMetadataBlocks={errorLoadingMetadataBlocksToRender}
          />
        )}
      </article>
      <NotImplementedModal show={isModalOpen} handleClose={hideModal} />
    </>
  )
```

Reply:
yes, makes sense